File: ReSharper.FSharp/src/Projects.FSharp/src/FSharpProjectPresenter.cs
using JetBrains.Platform.ProjectModel.FSharp.Properties;
using JetBrains.ProjectModel;
using JetBrains.ProjectModel.Resources;
using JetBrains.UI.Icons;

namespace JetBrains.Platform.ProjectModel.FSharp
{
  [ProjectModelElementPresenter(Priority)]
  public class FSharpProjectPresenter : IProjectModelElementPresenter
  {
    private const int Priority = 2; // needs to be greater than ProjectModelElementPresenter priority (which is 1)

    public IconId GetIcon(IProjectModelElement projectModelElement)
    {
      var project = projectModelElement as IProject;
      if (project?.ProjectProperties is FSharpProjectProperties)
        return ProjectModelThemedIcons.FsharpProject.Id;
      return null;
    }

    public string GetPresentableLocation(IProjectModelElement projectModelElement)
    {
      return null;
    }
  }
}

File: ReSharper.FSharp/src/FSharp/FSharp.Common/src/ProjectModel/FSharpProjectPresenter.fs
namespace JetBrains.ReSharper.Plugins.FSharp.ProjectModel

open JetBrains.ProjectModel
open JetBrains.ProjectModel.Resources
open JetBrains.ReSharper.Plugins.FSharp.ProjectModel.Scripts

[<ProjectModelElementPresenter(2.0)>]
type FSharpProjectPresenter() =
    interface IProjectModelElementPresenter with
        member x.GetIcon(projectModelElement) =
            match projectModelElement with
            | :? IProject as project when (project.ProjectProperties :? FSharpProjectProperties) ->
                ProjectModelThemedIcons.FsharpProject.Id
            | :? FSharpScriptModule ->
                ProjectModelThemedIcons.Fsharp.Id
            | _ -> null

        member x.GetPresentableLocation _ = null